src/RemoteSymbolProvider/MicrosoftSymbolServerSymbolProviderTest.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_format.h>
+#include <absl/strings/str_replace.h>
+#include <gtest/gtest.h>
+
+#include <QCoreApplication>
+
+#include "Http/MockDownloadManager.h"
+#include "OrbitBase/Result.h"
+#include "OrbitBase/StopSource.h"
+#include "OrbitGgp/MockClient.h"
+#include "QtUtils/MainThreadExecutorImpl.h"
+#include "RemoteSymbolProvider/MicrosoftSymbolServerSymbolProvider.h"
+#include "Symbols/MockSymbolCache.h"
+#include "TestUtils/TestUtils.h"
+
+namespace orbit_remote_symbol_provider {
+
+namespace {
+using orbit_base::Canceled;
+using orbit_base::CanceledOr;
+using orbit_base::Future;
+using orbit_ggp::SymbolDownloadInfo;
+using SymbolDownloadQuery = orbit_ggp::Client::SymbolDownloadQuery;
+using orbit_symbol_provider::ModuleIdentifier;
+using orbit_test_utils::HasError;
+using orbit_test_utils::HasNoError;
+using ::testing::_;
+using ::testing::Return;
+
+const std::filesystem::path kSymbolCacheDir{"symbol/cache/path"};
+const std::string kValidModuleName{"only_available_module_name"};
+const std::string kValidModuleBuildId{"ABCD12345678"};
+const ModuleIdentifier kValidModuleId{absl::StrFormat("module/path/to/%s", kValidModuleName),
+                                      kValidModuleBuildId};
+const std::string kValidModuleDownloadUrl{
+    absl::StrFormat("https://msdl.microsoft.com/download/symbols/%s/%s/%s", kValidModuleName,
+                    kValidModuleBuildId, kValidModuleName)};
+
+class MicrosoftSymbolServerSymbolProviderTest : public testing::Test {
+ public:
+  MicrosoftSymbolServerSymbolProviderTest()
+      : symbol_provider_{&symbol_cache_, &download_manager_},
+        executor_(orbit_qt_utils::MainThreadExecutorImpl::Create()) {
+    EXPECT_CALL(symbol_cache_, GenerateCachedFileName)
+        .WillRepeatedly([](const std::filesystem::path& module_file_path) {
+          auto file_name = absl::StrReplaceAll(module_file_path.string(), {{"/", "_"}});
+          return kSymbolCacheDir / file_name;
+        });
+  }
+
+  enum class DownloadResultState { kSuccess, kCanceled, kError };
+  void SetUpDownloadManager(DownloadResultState expected_state, std::string expected_url,
+                            std::string error_msg = "") {
+    EXPECT_CALL(download_manager_, Download)
+        .Times(1)
```

Comment:
I am not sure if this `.Times(1)` is needed if there is a `WillOnce`. But it also does not hurt

Reply:
It's true that it can be inferred if not mentioned explicitly go/googletest-mocking#EXPECT_CALL.Times

Reply:
Thanks for looking that up!